src/ape/types/trace.py

```diff
@@ -216,3 +222,26 @@ class TraceFrame(BaseInterfaceModel):
     """
     The number of external jumps away the initially called contract (starts at 0).
     """
+
+    raw: Dict
+    """
+    The raw trace frame data.
+    """
+
+
+class LineTraceNode(BaseModel):
+    source_id: str
+    """
+    The local project's contract's relative path str as an ID.
+    """
+
+    method_id: str
+    """
+    The method containing the lines.
+    """
+
+    lines: Dict[int, str]
```

Comment:
I need to understand the intent behind this better. I am not sure it's a 1:1 releationship in all cases.

Reply:
Line number -> Line content is 1 to 1

Reply:
Ah, so this is just the model of lines from a `ethpm_types.Source` object? Is there a way to expand more functionality there instead? maybe `Source.__getitem__`

Reply:
Ya there are probably a few things we can move to ethpm types!

I will explain more soon about these line traces soon... I am still kind of in a research phase